joomla/joomla-cms
# [4.0] rename Joomla core classNames fas fa- back to icon- (and still use Font Awesome)

Pull Request for Issue # .

### Summary of Changes

All looked well when @N6REJ implemented Font Awesome in Joomla Administrator. Trouble began when the magic words "backward compatibility" where mentioned. During Joomla 4 both Font Awesome and icons from 3th party extensions with the Joomla 3 prefix "icon-" should be loaded. 

As Font Awesome documentation states in the first line of _[Basic Use](https://fontawesome.com/how-to-use/on-the-web/referencing-icons/basic-use)_: 
> You can place Font Awesome icons just about anywhere using a style prefix and the icon’s name. 

A prefix and an icon name. 

By default this prefix is set to `fa`.
There is a variable in the Joomla Administrator template Atum where you can set the prefix. 
https://github.com/joomla/joomla-cms/blob/eb9f540453f76898cf5b0e86f7408179e6550885/administrator/templates/atum/scss/_variables.scss#L128
This prefix is set to `fa` and can be changed to whatever. 

To fix all 3th party backwards compatibility shizzle this PR replaces all instances of `fas fa-` by `icon-` and still uses FontAwesome. 

### Testing Instructions

- Browse through Joomla Administrator and notice the presence of icons. 
- Install Akeeba and / or JCE and see their icons too. 
- Apply this PR
- Run `npm run build:css`
- Repeat the first step and browse through Joomla Administrator and notice the presence of icons.  
  - even the icons for Akeeba and / or JCE are present. 

### Actual result BEFORE applying this Pull Request

<img width="565" alt="Schermafbeelding 2020-09-21 om 08 38 45" src="https://user-images.githubusercontent.com/639822/93738365-e4077d00-fbe5-11ea-9d15-c4a2e0d10c82.png">

### Expected result AFTER applying this Pull Request

<img width="567" alt="Schermafbeelding 2020-09-21 om 08 25 17" src="https://user-images.githubusercontent.com/639822/93738370-e5d14080-fbe5-11ea-9bcc-95b2dd9eaf83.png">

### Documentation Changes Required

I do think so... :-) But I don't know where. Not familiar with the Joomla Font Awesome documentation. Sorry
